### PR TITLE
Remove python 2 legacy from download_manager.py

### PR DESCRIPTION
**Pull request by Eshan-Agarwal**
As python 2 use is been dropped. So Remove unnecessary code from `download_manager.py` .

@Conchylicultor please have a look

### CODE REVIEW

File: tensorflow_datasets/core/download/download_manager.py [hunk 1]
```diff
@@ -389,29 +389,7 @@ def manual_dir(self):
...
 
-# ============================================================================
-# In Python 2.X, threading.Condition.wait() cannot be interrupted by SIGINT,
-# unless it's given a timeout. Here we artificially give a long timeout to
-# allow ctrl+C.
-# This code should be deleted once python2 is no longer supported.
-if sys.version_info[0] > 2:
-
-  def _wait_on_promise(p):
-    return p.get()
```
Comment (Conchylicultor) [inline tensorflow_datasets/core/download/download_manager.py]: You still need to resolve the promise. The code in `sys.version_info[0] > 2` is executed for Python3 so it should be kept.

Comment (Eshan-Agarwal) [reply tensorflow_datasets/core/download/download_manager.py]: Okay will make changes. Can you please tell me what is p here I didn't get it.

File: tensorflow_datasets/core/download/download_manager.py [hunk 2]
```diff
@@ -388,8 +388,14 @@ def manual_dir(self):
               self._manual_dir, self._manual_dir_instructions))
     return self._manual_dir
 
+  
+if sys.version_info[0] > 2:
```
Comment (Conchylicultor) [inline tensorflow_datasets/core/download/download_manager.py]: Now that this is always true, the condition can be removed.

Comment (Eshan-Agarwal) [reply tensorflow_datasets/core/download/download_manager.py]: Yeah, I was also thinking of it . Thanks

File: tensorflow_datasets/core/download/download_manager.py (hunk 1 above)
Comment (Conchylicultor) [reply tensorflow_datasets/core/download/download_manager.py]: `p` means `promise` from https://pypi.org/project/promise/

**Pull request merged**